src/lhttpc_lb.erl

```diff
@@ -39,13 +43,16 @@ checkout(Host, Port, Ssl, MaxConn, ConnTimeout) ->
     Lb = find_lb({Host,Port,Ssl}, {MaxConn, ConnTimeout}),
     gen_server:call(Lb, {checkout, self()}, infinity).
 
-%% Called when the socket has died and we're done
--spec checkin(host(), port_number(), SSL::boolean()) -> ok.
-checkin(Host, Port, Ssl) ->
-    case find_lb({Host,Port,Ssl}) of
-        {error, undefined} -> ok; % LB is dead. Pretend it's fine -- we don't care
-        {ok, Pid} -> gen_server:cast(Pid, {checkin, self()})
-    end.
+%% Returns the LB state
```

Comment:
Alright. If it could have a docline for this, it would be good.

The thing that would for example worry me is have people depend on it programmatically in production; I'm not sure what the effect of a fold would be on performance over these tables given the fold will force the usage of an ets:fixtable or may just hang the table for a while.

If we want a more general usage, I'd modify it so the function reads all the data it can from the table in one shot, and *then* polls all the load balancers. This limits the interaction with the table to the shortest amount of time possible, whereas calling busy load balancers can take many milliseconds to return.